lib/floe/validation_mixin.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Floe
+  module ValidationMixin
+    def self.included(base)
+      base.extend(ClassMethods)
+    end
+
+    def parser_error!(comment)
+      self.class.parser_error!(full_name_string, comment)
+    end
+
+    def parser_missing_field!(field_name)
+      self.class.parser_missing_field!(full_name_string, field_name)
+    end
+
+    def parser_invalid_field!(field_name, field_value = nil, comment = nil)
+      self.class.parser_invalid_field!(full_name_string, field_name, field_value, comment)
+    end
+
+    def workflow_state?(field_value, workflow)
+      workflow.payload["States"] ? workflow.payload["States"].include?(field_value) : true
+    end
+
+    private
+
+    def full_name_string
+      full_name.join(".")
+    end
+
+    module ClassMethods
+      def parser_error!(full_name, comment)
+        full_name = full_name.join(".") if full_name.kind_of?(Array)
```

Comment:
Do we need this and also `#full_name_string`?  Could we enforce one or the other or are there some cases where we have to pass an array in?

Reply:
Good call. Workflow is a little funky with this, but the code seems fine dropping this extra concept